TS_1/task2.2.ts

```diff
@@ -0,0 +1,63 @@
+// 2. Создайте функцию, которая подсчитает, сколько в объекте значений каждого типа. 
+//   Принимает на вход объект или массив таких же объектов, у которого ключ всегда string, а значение - string, number, boolean. 
+//   Возвращает же - объект с ключами string, number, boolean и количеством таких значений в объекте или в сумме у всех объектов в массиве.
+
+type ValueType = string | number | boolean;
+
+interface ICountResult {
+    string: number;
+    number: number;
+    boolean: number;
+}
+
+function countValueTypes(input: Record<string, ValueType> | Record<string, ValueType>[]): ICountResult {
+    const result: ICountResult = {
+        string: 0,
+        number: 0,
+        boolean: 0
+    };
+
+    const countTypes = (obj: Record<string, ValueType>): void => {
```

Comment:
You can move this function outside and use as helper, but not neccessary :)

Reply:
Definetely it is necessary. Especially when function is connected to parent function's "state", i mean result variable. helper functions should not have any dependency on such variables and should return calculated data

Reply:
I see